bindings/capacitorjs/ios/Plugin/Plugin.swift

```diff
@@ -0,0 +1,207 @@
+                "secretManager":{
+                    "stronghold":{
+                        "snapshotPath":"\(path)/wallet.stronghold"
+                }}
+            }
+            """
+            print(options)
+            
+            // TODO: implement logger as a fn
+            let filename = "\(path)/iota_wallet.log"
+            let level_filter = "debug"
+            iota_init_logger(filename.cString(using: .utf8), level_filter.cString(using: .utf8))
+            
+            // Keep the C++ handler / pointer of the messageHandler call result
+            let handler: OpaquePointer? = iota_initialize(options, error_buffer, error_buffer_size)
+            // Convert pointer to integer keeping bit pattern
+            call.resolve(["messageHandler": Int(bitPattern: handler)])
+        
+        } catch {
+            call.reject("failed to initialize messageHandlerNew")
+        }
+    }
+
+    @objc func destroy(_ call: CAPPluginCall) {
+        print("Capacitor call destroy received", call.jsObjectRepresentation)
+        guard let handler = call.getInt("messageHandler") else {
+            return call.reject("handler is required")
+        }
+        iota_destroy(OpaquePointer(bitPattern: handler))
+        call.resolve()
+    }
+
+    @objc func sendMessage(_ call: CAPPluginCall) {
+        guard let handler = call.getInt("handler") else {
+            return call.reject("handler is required")
+        }
+        let messageHandler: OpaquePointer? = OpaquePointer(bitPattern: handler)
+        
+        // replacing for urls slashes since it's serialized on JS
+        guard let message = call.getString("message")?
+                .replacingOccurrences(of: "\\", with: "") else {
+            return call.reject("message is required")
+        }
+        
+        // the object to be passed as a context data on their callback
+        let contextResult = ContextResult(_call: call)
+        // context var to be passed on the object callback
+        // where it will be converted back to object type ready to use
+        let context = Unmanaged<ContextResult>.passRetained(contextResult).toOpaque()
+        
+        iota_send_message(messageHandler, message, contextResult.callback, context)
+    }
+
+    @objc func listen(_ call: CAPPluginCall) {
+        print("Capacitor call listen received", call.jsObjectRepresentation)
+        guard let handler = call.getInt("messageHandler") else {
+            return call.reject("handler is required")
+        }
+        let messageHandler: OpaquePointer? = OpaquePointer(bitPattern: handler)
+        guard let eventTypes = call.getArray("eventTypes") else {
+            return call.reject("eventTypes is required")
+        }
+        
+        let contextResult = ContextResult(_call: call)
+        let context = Unmanaged<ContextResult>.passRetained(contextResult).toOpaque()
+        
+        let error_buffer: UnsafeMutablePointer<CChar>? = nil
+        let error_buffer_size = 0
```

Comment:
These are already defined on lines 52 and 53, but now I wonder if we should be defining them in each function instead of outside the functions since the error buffer probably shouldn't be held onto after we do `call.resolve()`  🤔

Reply:
Added inside each function, thanks!

Reply:
thanks! have just fixed it.